src/pages/settings/Profile/ProfilePage.js

```diff
@@ -67,55 +66,28 @@ class ProfilePage extends Component {
+        this.logins = this.getLogins(props.loginList);
+        this.avatar = {uri: lodashGet(this.props.currentUserPersonalDetails, 'avatar', ReportUtils.getDefaultAvatar(this.props.currentUserPersonalDetails.login))};
+        this.pronouns = props.currentUserPersonalDetails.pronouns;
         this.state = {
-            firstName: this.props.currentUserPersonalDetails.firstName,
-            hasFirstNameError: false,
-            lastName: this.props.currentUserPersonalDetails.lastName,
-            hasLastNameError: false,
-            pronouns: this.props.currentUserPersonalDetails.pronouns,
-            hasPronounError: false,
-            hasSelfSelectedPronouns: !_.isEmpty(this.props.currentUserPersonalDetails.pronouns) && !this.props.currentUserPersonalDetails.pronouns.startsWith(CONST.PRONOUNS.PREFIX),
-            selectedTimezone: lodashGet(this.props.currentUserPersonalDetails.timezone, 'selected', CONST.DEFAULT_TIME_ZONE.selected),
-            isAutomaticTimezone: lodashGet(this.props.currentUserPersonalDetails.timezone, 'automatic', CONST.DEFAULT_TIME_ZONE.automatic),
-            logins: this.getLogins(props.loginList),
-            avatar: {uri: lodashGet(this.props.currentUserPersonalDetails, 'avatar', ReportUtils.getDefaultAvatar(this.props.currentUserPersonalDetails.login))},
+            isAutomaticTimezone: lodashGet(props.currentUserPersonalDetails.timezone, 'automatic', CONST.DEFAULT_TIME_ZONE.automatic),
+            hasSelfSelectedPronouns: !_.isEmpty(props.currentUserPersonalDetails.pronouns) && !props.currentUserPersonalDetails.pronouns.startsWith(CONST.PRONOUNS.PREFIX),
             isAvatarChanged: false,
         };
 
         this.getLogins = this.getLogins.bind(this);
-        this.setAutomaticTimezone = this.setAutomaticTimezone.bind(this);
+        this.validate = this.validate.bind(this);
         this.updatePersonalDetails = this.updatePersonalDetails.bind(this);
-        this.validateInputs = this.validateInputs.bind(this);
         this.updateAvatar = this.updateAvatar.bind(this);
     }
 
     componentDidUpdate(prevProps) {
-        let stateToUpdate = {};
-
         // Recalculate logins if loginList has changed
-        if (this.props.loginList !== prevProps.loginList) {
-            stateToUpdate = {...stateToUpdate, logins: this.getLogins(this.props.loginList)};
-        }
-
-        if (_.isEmpty(stateToUpdate)) {
+        if (this.props.loginList === prevProps.loginList) {
             return;
         }
-
-        // eslint-disable-next-line react/no-did-update-set-state
-        this.setState(stateToUpdate);
-    }
-
-    /**
-     * Set the form to use automatic timezone
-     *
-     * @param {Boolean} isAutomaticTimezone
-     */
-    setAutomaticTimezone(isAutomaticTimezone) {
-        this.setState(({selectedTimezone}) => ({
-            isAutomaticTimezone,
-            selectedTimezone: isAutomaticTimezone ? moment.tz.guess() : selectedTimezone,
```

Comment:
Thanks for fixing the timezone issue. However, I think it still doesn't quite work as it did before. If we manually select a different timezone than our own and then check `Set my timezone automatically`, we no longer guess the new timezone and update it accordingly.

Reply:
> I'm sorry - maybe I'm out of sync. Is there anything pending with this merge request please?

@liyamahendra yep, you need to address @luacmartins' comments and also fix the merge conflicts

Reply:
Hey guys, while the initial fix for the reported issue was a small modification of adding a prop to scrollview, the scope of this got expanded to refactor the Profile Screen and use Form.

I have already spent a lot of time on this. With all due respect, want to ask - is there a chance to review & revise the compensation for this task?

@rushatgabhane @luacmartins

Reply:
@liyamahendra I understand, I'll ask about it internally

Just to be on the same page, you agreed to take up the task of refactoring profile page to use Form. https://github.com/Expensify/App/issues/9234#issuecomment-1151524666